app/helpers.py

```diff
@@ -286,46 +286,54 @@ def get_round(
 
 
 def get_feedback_survey_data(
-    application, application_id, current_feedback_list, section_display_config
+    application,
+    application_id,
+    current_feedback_list,
+    section_display_config,
+    feedback_survey_config,
 ):
 
-    # we grab the number from the last section i.e "6. Foobar" then increment it to get "7. " for feedback.
-    last_section_title = section_display_config[-1].title
-    number = None
-    if re.search(r"\b(\d+)\.", last_section_title):
-        number = int(re.search(r"\b(\d+)\.", last_section_title).group(1)) + 1
-
-    round_feedback_available = application.all_forms_complete and all(
-        current_feedback_list
-    )
-    existing_survey_data_map = {
-        page_number: get_survey_data(application_id, page_number)
-        for page_number in ["1", "2", "3", "4"]
-    }
+    if feedback_survey_config.requires_survey:
+        # we grab the number from the last section i.e "6. Foobar" then increment it to get "7. " for feedback.
+        last_section_title = section_display_config[-1].title
+        number = None
+        if re.search(r"\b(\d+)\.", last_section_title):
+            number = (
+                int(re.search(r"\b(\d+)\.", last_section_title).group(1)) + 1
+            )
```

Comment:
Was going to complain about this code then realised I wrote it, and this PR just changed indentation 😂 
```suggestion
        _match = re.search(r"\b(\d+)\.", last_section_title)
        number = int(_match.group(1)) + 1 if match else None
```

Reply:
Do you want me to commit the suggestion ?

Reply:
Up to you!